components/builder-api-client/src/builder.rs

```diff
@@ -462,6 +462,33 @@ impl BuilderAPIProvider for BuilderAPIClient {
                       progress)
     }
 
+    /// Create an origin
+    ///
+    ///  # Failures
+    ///
+    ///  * Remote builder is not available
+    ///  * Origin already exists
+    ///  * Y'aint got no auth
+    fn create_origin(&self, origin: &str, token: &str) -> Result<()> {
+        let body = json!({
+            "name": origin,
+        });
+
+        let sbody = serde_json::to_string(&body)?;
+
+        let res = self.add_authz(self.0.post("depot/origins"), token)
+                      .body(&sbody)
+                      .header(Accept::json())
+                      .header(ContentType::json())
+                      .send()?;
+
+        if res.status != StatusCode::Created {
+            return Err(err_from_response(res));
+        }
+
+        Ok(())
```

Comment:
I find code a bit simpler to understand when
1. The common/happy case comes first
1. The conditions are positive rather than negated
1. Explicit returns are avoided

Sometimes these conflict, but in this case, I think we can have them all and it leads to a structure that I find more pleasing:
```rust
        if res.status == StatusCode::Created {
            Ok(())
        } else {
            Err(err_from_response(res))
        }
```

Reply:
I'll fix up some of this in my PR after this merges... although I decided to go with the negated case first as it ended up reading a bit cleaner to me